File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.java
import java.io.*;
import java.util.*;

public class Main {
    static boolean isMo(char c){
        return c == 'a' || c== 'i' || c =='o' || c == 'e' || c == 'u';
    }
    static boolean check(String s){
        char[] crr = s.toCharArray();
        boolean flag1 = false; //모음존재체크
        boolean flag2 = true;//자모음3개연속 체크
        boolean flag3 = true; // 같은글자연속체크
        int cntJ = 0, cntM= 0;
        char prev =  ' ' ;

        for(char c : crr){
            if(c == prev && (c != 'e' && c != 'o')) flag3 = false;
            if(isMo(c)){
                cntM++; cntJ = 0; flag1 = true;
            } else {
                cntJ++; cntM = 0;
            }
            if(cntM >= 3 || cntJ >= 3) flag2 = false;
            prev = c;
        }

        return flag1 && flag2 && flag3;
    }
    public static void main(String[] args) throws IOException {
        var in = new BufferedReader(new InputStreamReader(System.in));
        var out = new PrintWriter(System.out);

        while(true){
            String s = in.readLine();
            if(s.equals("end")) break;
            if(check(s)){
                out.printf("<%s> is acceptable.\n", s);
            } else {
                out.printf("<%s> is not acceptable.\n", s);
            }
        }


        out.flush();
        out.close();
    }
}

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
def isMo(c):
    return c == 'a' or c == 'e' or c =='o' or c=='i' or c == 'u'
while True:
    s = input()
    if s == 'end': break
    cnt_j, cnt_m, prev, cnt = 0, 0, '', 0
    flag = 0
    for c  in s:
        if isMo(c): 
            cnt_m += 1 
            cnt += 1
            cnt_j = 0
        else: 
            cnt_j += 1
            cnt_m = 0
        if cnt_j >= 3 or cnt_m >= 3:
            flag = 1
            break
        if prev != '' and c != 'e' and c != 'o' and prev == c:
            flag = 1
            break
        prev = c
    if flag:
        print(f"<{s}> is not acceptable.")
    else:
        if cnt == 0:
            print(f"<{s}> is not acceptable.")
        else:
            print(f'<{s}> is acceptable.')